Promote sprite depth examples from experimental to main

pushfoo opened the pull request:
1. Stop trying to set the window's read-only clock value
2. Convert them to the new `View` example style
3. Add entries for them in the examples listing

pushfoo committed:
Fix sprite depth example clock and style

* Stop trying to set the window's read-only clock value

* Fix annoying indent on list at top of comments

DragonMoffon commented on arcade/experimental/sprite_depth_cosine.py:
```diff
@@ -88,11 +87,10 @@ def on_key_press(self, symbol: int, modifiers: int):
-        self.time += delta_time
-
         for i, sprite in enumerate(self.sprite_list):
+            # time ticks automatically now due to Window's global clock
```
We have to assume most users will have never looked at this example before so this comment doesn't make sense. (Also this might not have existed pre 3.0 so even less likely to have been seen before)

pushfoo replied:
Removed.

DragonMoffon reviewed (changes requested):
The window has a depth buffer by default, and spritelists innately support depth testing so this should be moved to `examples`

pushfoo committed:
Convert sprite_depth_cosine.py to use Views style

* Make the example subclass view

* Use the same style found in other examples

* Grab the context locally

pushfoo committed:
Move sprite depth cosine example

pushfoo committed:
Fix run string at top of file

alejcas commented:
Seems like this line is wrong:

`if self.use_depth:`

You moved `use_depth` to the window so the line should check:

`if self.window.use_depth:`

Anyway I don't understand why you moved that attribute to the window. Just leave it in the view...no?

pushfoo committed:
Convert depth of field example to View style

* Convert to View style

* Make bg / clear color a Color defined at the top of the file

* Use WINDOW_TITLE, WINDOW_WIDTH, and WINDOW_HEIGHT constants

alejcas commented:
> Seems like this line is wrong:
> 
> `if self.use_depth:`
> 
> You moved `use_depth` to the window so the line should check:
> 
> `if self.window.use_depth:`
> 
> Anyway I don't understand why you moved that attribute to the window. Just leave it in the view...no?

I think this is what's wrong:

`window.use_depth = not window.use_depth`

pushfoo committed:
Move depth of field example into main examples

pushfoo committed:
Update run string at top of file

pushfoo committed:
Add sprite_depth_buffer to the examples page

* Add a screenshot for the example page and example gallery

* Add an .rst file to reference the code and example

* Add the .rst file to the examples gallery

pushfoo committed:
Add depth_of_field.py to the examples gallery

* Add a screenshot for the page and gallery

* Add an example .rst file for it

* Add an entry in the examples page

pushfoo committed:
Fix DoF example run line

pushfoo committed:
Fix issue jans pointed out

pushfoo commented:
@alejcas fixed

pushfoo committed:
Remove unused assignment

pushfoo committed:
Use black bg for sprite_depth_cosine screenshot

pushfoo committed:
Put example tiles in appropriate sections based on feedback

pushfoo committed:
Move the sprite depth example to the Sprite Properties heading

DragonMoffon commented on arcade/examples/depth_of_field.py:
```diff
@@ -171,7 +178,7 @@ def render(self):
 
 
-class App(Window):
+class App(View):
```
convention is `GameView`. Even though it's not really accurate, the examples used to be a mess in this regard, so im forcing a naming convention.

DragonMoffon commented on arcade/examples/depth_of_field.py:
```diff
@@ -234,11 +242,26 @@ def on_draw(self):
 
-        # Draw the blurred frame buffer and then the focus display
-        self.use()
+        # Draw the blurred frame buffer and then the focus display\
```
accidental slash

DragonMoffon commented on arcade/examples/sprite_depth_cosine.py:
```diff
@@ -23,21 +23,22 @@
 
 
-class MyGame(arcade.Window):
+class DepthExample(arcade.View):
```
Same naming note as above. post 3.0, we could discuss a different convention, but for now...

DragonMoffon commented on arcade/examples/sprite_depth_cosine.py:
```diff
@@ -88,11 +89,26 @@ def on_key_press(self, symbol: int, modifiers: int):
     def on_update(self, delta_time):
-        self.time += delta_time
-
+        # We absolute time instead of the delta, so we get it from the window's clock
```
This might be github, but `we absolute time` doesn't make sense.

pushfoo committed:
Fix comments and naming conventions per GitHub review

pushfoo committed:
Tweak phrasing

einarf commented:
The reason it wasn't moved into examples was a lingering issue related to the depth range. As far as I know the depth range set in the ortho projection is `[-1.0, 1.0]` defining the `depth` that can be set in the sprite's geometry.

We could possibly change to to `[-1, 100]` to give the user a more reasonable range to work with? If the sprite depth is set outside this range it risks being clipped in primitive assembly not appearing.

Including a new reasonable depth range would make things a bit more cleaner?

einarf commented:
It seems @DragonMoffon changed the range to `[-100, 100]` so we should make sure to mention this range in the example and take advantage of this range if that's not already in place. Seems it was modified a bit from the original -1 to 1 range.

The default depth range in arcade is -100 (near plane) / 100 (far plane). Higher value is further away from the camera.

pushfoo commented:
Can we add default near and far constants to make this more legible?

einarf commented:
For now I think it's enough to just mention the default range.

DragonMoffon commented:
I didn't actually choose this range myself. There was definitely a code snippet in Arcade that I based it on. I feel like it was the internal matrix gen, but it might have been one of the gl tutorials.

In any case, in the depth example, it should probably be mentioned since it's directly settable in Camer2D's init

DragonMoffon commented on arcade/examples/sprite_depth_cosine.py:
```diff
@@ -88,11 +89,26 @@ def on_key_press(self, symbol: int, modifiers: int):
     def on_update(self, delta_time):
-        self.time += delta_time
-
+        # We use absolute time instead of the delta, so we get it from the window's clock
```
I would still change this to 'elapsed' since that's the language used by the clocks. It is also missing a 'the'

'We use the elapsed time instead of the delta, so we get it from the window's clock'

einarf replied:
Just use the time from the clock. There are several examples can could need this change.

DragonMoffon reviewed (approved):
One last, very pedantic, change. Im otherwise happy for it to be merged.

pushfoo committed:
Update phrasing further

pushfoo committed:
Centralize and document near/far defaults a bit

pushfoo committed:
Linter

pushfoo committed:
More linter

pushfoo commented:
- [x] Added doc about near / far clipping
- [x] Changed time phrasing to be lessspecific to American / Hemmingway style
- [x] Centralized near / far defaults from a single source of truth

pushfoo commented:
> ["yeah looks good" - DragonMoffon](https://discord.com/channels/458662222697070613/458662458198982676/1299560812221825054) (Discord)
